src/test/scala/com/microsoft/hyperspace/index/IndexLogEntryTest.scala

```diff
@@ -362,40 +362,26 @@ class IndexLogEntryTest extends HyperspaceSuite with SQLHelper {
     assert(directoryEquals(actual, expected))
   }
 
-  test("Directory.fromLeafFiles: throwIfNotExist flag throws exception for non-existent" +
-    "directory, otherwise works as expected.") {
-    val testDirPath = toPath(testDir)
-    val nonExistentDir = new Path(testDirPath, "nonexistent")
-
-    // Try create Directory object with throwIfNotExists to true. This should throw exception.
-    intercept[FileNotFoundException] {
-      Directory.fromDirectory(nonExistentDir, fileIdTracker, throwIfNotExists = true)
-    }
-
-    // Try create Directory object with throwIfNotExists to false. This should create empty
-    // Directory.
-    val expected = {
-      val nonExistentDirDirectory = Directory(nonExistentDir.getName)
-      createDirectory(nonExistentDir, nonExistentDirDirectory)
-    }
-
-    val actual = Directory.fromDirectory(nonExistentDir, fileIdTracker, throwIfNotExists = false)
-    assert(directoryEquals(actual, expected))
-  }
-
-  test("Directory.fromDirectory where the directory is empty.") {
+  test("Directory.fromDirectory where the directory is empty or nonexistent.") {
     val testDirPath = toPath(testDir)
     val emptyDirPath = new Path(testDirPath, "empty")
-
-    // Try create Directory object with throwifNotExists to false. This should create empt
-    // Directory.
     val expected = {
       val emptyDirDirectory = Directory(emptyDirPath.getName)
       createDirectory(emptyDirPath, emptyDirDirectory)
     }
 
-    val actual = Directory.fromDirectory(emptyDirPath, fileIdTracker)
-    assert(directoryEquals(actual, expected))
+    {
+      // Test non-existent directory.
+      val actual = Directory.fromDirectory(emptyDirPath, fileIdTracker)
```

Comment:
Is the directory really non-existent? It seems it is created at line 369.

Reply:
ditto it's also `Directory` object